src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-utils.c

```diff
@@ -377,8 +380,17 @@ long xclmgmt_hot_reset(struct xclmgmt_dev *lro, bool force)
 	xocl_clear_pci_errors(lro);
 	store_pcie_link_info(lro);
 
-	/* Update the userspace fdt with the current values in the mgmt driver */
+	/*
+	 * Update the userspace fdt with the current values in the mgmt driver
+	 *
+	 * For vmr supported versal devices, we enabled A/B boot, thus we should
+	 * reload fdt from the right boot image instead of using unchanged fdt
+	 * for other ALEVO devices.
+	 */
+	mutex_lock(&lro->busy_mutex);
+	(void) xclmgmt_reload_fdt_blob_vmr(lro);
```

Comment:
I have 2 questions here
1. since you are going to reload fdt, meaning destroy and recreate all subdevices for vmr case, then those online operations in this reset func are still required? functionality wise, they can be kept, just wonder whether they are necessary
2. you rely on the xgq info from the previous metadata to get new metadata. Is there a case that A/B boot also changes the xgq info?  getting xgq info from vsec should be always reliable

Reply:
1. On pf0, we only offline and online some subdevs which are not affected by the fdt. Then we reload fdt_blob, shared it with pf1.
2.The concern is that the VSEC info should be correct so that xgq_vmr subdev will be probed successfully. We always create xgq_vmr subdev from VSEC directly, code is here: xocl_subdev_create_vsec_devs.

Reply:
1. reload fdt will destroy all subdevices and recreate them again. my asking is since we are going to destroy them, why we bother to online them first
2. xclmgmt_probe will create xgq_vmr first from vsec and recreate it with the info from metadata. for reset, the xgq_vmr is from the info of metadata of the old one, which is onlined from previous step, not vsec, you use the xgq to get metadata and recreate xgq again.

Reply:
1. is this related to this PR? Sorry, I don't follow what your are asking. I don't change the behavior I hope.
2. There is a case where shell pkg is missing or not matching the VSEC, thus loading firmware will not continue. We still want the xgq_vmr work for flashing the shell. For reset, we should support flashing the shell from backup image and block other harmful operations. I think loading from the metadata from the shell is the right way. Unless I am missing some other corner cases?

Reply:
ok.
1. it is my bad that I misread the code
2. it is a different issue that would arise when xgq in A and B are not compatible. That should be tracked by different CR

Reply:
@xuhz filed CR https://jira.xilinx.com/browse/CR-1147030